aten/src/ATen/native/cuda/MultinomialKernel.cu

```diff
@@ -283,7 +282,9 @@ sampleMultinomialOnce(int64_t* dest,
       __syncthreads();
     }
 
-    if (threadIdx.x == 0 && !found) {
+    if (threadIdx.x == 0 && found) {
+        dest[curDist] = foundPos;
+    } else if (threadIdx.x == 0 && !found) {
```

Comment:
nit: `threadIdx.x == 0` only needs to be checked once, right?

Reply:
Yes, you are right. I will change that!